packages/express-server/middleware/userIdMiddleware.ts

```diff
@@ -0,0 +1,17 @@
+import express from 'express';
+import { v4 } from 'uuid';
+
+export const userIdMiddleware = (
+  req: express.Request,
+  res: express.Response,
+  next: Function
+): void => {
+  let userId: string | undefined = req.cookies.userId;
+  if (!userId) {
+    userId = v4();
+    res.cookie('userId', userId);
+    req.cookies.userId = userId;
```

Comment:
i would put the userId on the req.
i.e - if there's no user id, write the cookie and put it on the req, else read the cookie and put it on the req.

then u handle the cookies only here